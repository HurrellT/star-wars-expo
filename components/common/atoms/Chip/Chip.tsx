import React from 'react';
import { View } from 'react-native';
import { Text } from '~/components/ui/text';
import { 
  ChipColor, 
  ChipSize, 
  ChipVariant, 
  getColorClasses, 
  getTextColorClasses,
  getSizeClasses, 
  getBorderClass 
} from './Chip.styles';

type ChipProps = {
  children: React.ReactNode;
  color?: ChipColor;
  size?: ChipSize;
  variant?: ChipVariant;
  className?: string;
};

const Chip = ({ 
  children, 
  color = "default", 
  size = "md", 
  variant = "flat",
  className = ""
}: ChipProps) => {
  const colorClasses = getColorClasses(color, variant);
  const textColorClasses = getTextColorClasses(color, variant);
  const sizeClasses = getSizeClasses(size);
  const borderClass = getBorderClass(variant);

  return (
    <View className={`self-start ${sizeClasses} ${colorClasses} ${borderClass} ${className}`}>
      <Text className={`font-medium ${textColorClasses}`}>
        {children}
      </Text>
    </View>
  );
};

export default Chip;
