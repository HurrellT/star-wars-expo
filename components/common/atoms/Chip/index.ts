export { default } from './Chip';
export type { ChipColor, ChipSize, ChipVariant } from './Chip.styles';
export { getColorClasses, getTextColorClasses, getSizeClasses, getBorderClass } from './Chip.styles';
