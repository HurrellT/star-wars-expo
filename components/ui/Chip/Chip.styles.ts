export type ChipColor = "primary" | "secondary" | "success" | "warning" | "danger" | "default" | "muted";
export type ChipSize = "sm" | "md" | "lg";
export type ChipVariant = "flat" | "solid" | "bordered" | "ghost";

export const getColorClasses = (color: ChipColor, variant: ChipVariant): string => {
  const colorMap = {
    primary: {
      flat: 'bg-blue-500/20 border-blue-500/30',
      solid: 'bg-blue-500 border-blue-500',
      bordered: 'bg-transparent border-blue-500',
      ghost: 'bg-transparent hover:bg-blue-500/20',
    },
    secondary: {
      flat: 'bg-purple-500/20 border-purple-500/30',
      solid: 'bg-purple-500 border-purple-500',
      bordered: 'bg-transparent border-purple-500',
      ghost: 'bg-transparent hover:bg-purple-500/20',
    },
    success: {
      flat: 'bg-green-500/20 border-green-500/30',
      solid: 'bg-green-500 border-green-500',
      bordered: 'bg-transparent border-green-500',
      ghost: 'bg-transparent hover:bg-green-500/20',
    },
    warning: {
      flat: 'bg-yellow-500/20 border-yellow-500/30',
      solid: 'bg-yellow-500 border-yellow-500',
      bordered: 'bg-transparent border-yellow-500',
      ghost: 'bg-transparent hover:bg-yellow-500/20',
    },
    danger: {
      flat: 'bg-red-500/20 border-red-500/30',
      solid: 'bg-red-500 border-red-500',
      bordered: 'bg-transparent border-red-500',
      ghost: 'bg-transparent hover:bg-red-500/20',
    },
    muted: {
      flat: 'bg-gray-500/20 border-gray-500/30',
      solid: 'bg-gray-500 border-gray-500',
      bordered: 'bg-transparent border-gray-500',
      ghost: 'bg-transparent hover:bg-gray-500/20',
    },
    default: {
      flat: 'bg-slate-500/20 border-slate-500/30',
      solid: 'bg-slate-500 border-slate-500',
      bordered: 'bg-transparent border-slate-500',
      ghost: 'bg-transparent hover:bg-slate-500/20',
    },
  };

  return colorMap[color][variant] || colorMap.default.flat;
};

export const getTextColorClasses = (color: ChipColor, variant: ChipVariant): string => {
  const textColorMap = {
    primary: {
      flat: 'text-blue-700 dark:text-blue-200',
      solid: 'text-white',
      bordered: 'text-blue-700 dark:text-blue-200',
      ghost: 'text-blue-700 dark:text-blue-200',
    },
    secondary: {
      flat: 'text-purple-700 dark:text-purple-200',
      solid: 'text-white',
      bordered: 'text-purple-700 dark:text-purple-200',
      ghost: 'text-purple-700 dark:text-purple-200',
    },
    success: {
      flat: 'text-green-700 dark:text-green-200',
      solid: 'text-white',
      bordered: 'text-green-700 dark:text-green-200',
      ghost: 'text-green-700 dark:text-green-200',
    },
    warning: {
      flat: 'text-yellow-700 dark:text-yellow-200',
      solid: 'text-white',
      bordered: 'text-yellow-700 dark:text-yellow-200',
      ghost: 'text-yellow-700 dark:text-yellow-200',
    },
    danger: {
      flat: 'text-red-700 dark:text-red-200',
      solid: 'text-white',
      bordered: 'text-red-700 dark:text-red-200',
      ghost: 'text-red-700 dark:text-red-200',
    },
    muted: {
      flat: 'text-gray-700 dark:text-gray-200',
      solid: 'text-white',
      bordered: 'text-gray-700 dark:text-gray-200',
      ghost: 'text-gray-700 dark:text-gray-200',
    },
    default: {
      flat: 'text-slate-700 dark:text-slate-200',
      solid: 'text-white',
      bordered: 'text-slate-700 dark:text-slate-200',
      ghost: 'text-slate-700 dark:text-slate-200',
    },
  };

  return textColorMap[color][variant] || textColorMap.default.flat;
};

export const getSizeClasses = (size: ChipSize): string => {
  const sizeMap = {
    sm: 'px-2 py-1 text-xs rounded-full',
    md: 'px-3 py-1.5 text-sm rounded-full',
    lg: 'px-4 py-2 text-base rounded-full',
  };

  return sizeMap[size] || sizeMap.md;
};

export const getBorderClass = (variant: ChipVariant): string => {
  return variant === 'bordered' ? 'border' : '';
};
