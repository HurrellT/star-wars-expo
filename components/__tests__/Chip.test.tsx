import React from 'react';
import { render } from '@testing-library/react-native';
import Chip from '../common/atoms/Chip/Chip';

// Mock the problematic dependencies
jest.mock('@rn-primitives/slot', () => ({
  Slot: ({ children }: { children: React.ReactNode }) => children,
  Root: ({ children }: { children: React.ReactNode }) => children,
}));

jest.mock('~/components/ui/text', () => {
  const { Text: RNText } = jest.requireActual('react-native');
  return {
    Text: ({ children, className, ...props }: any) => (
      <RNText className={className} {...props}>
        {children}
      </RNText>
    ),
  };
});

describe('Chip Component', () => {
  it('renders correctly with default props', () => {
    const tree = render(<Chip>Default Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with flat variant', () => {
    const tree = render(<Chip variant="flat">Flat Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with solid variant', () => {
    const tree = render(<Chip variant="solid">Solid Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with bordered variant', () => {
    const tree = render(<Chip variant="bordered">Bordered Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with ghost variant', () => {
    const tree = render(<Chip variant="ghost">Ghost Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with primary color', () => {
    const tree = render(<Chip color="primary">Primary Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with secondary color', () => {
    const tree = render(<Chip color="secondary">Secondary Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with danger color', () => {
    const tree = render(<Chip color="danger">Danger Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with small size', () => {
    const tree = render(<Chip size="sm">Small Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with large size', () => {
    const tree = render(<Chip size="lg">Large Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom className', () => {
    const tree = render(<Chip className="custom-chip">Custom Chip</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with all props combined', () => {
    const tree = render(
      <Chip variant="solid" color="primary" size="lg" className="combined-chip">
        Combined Props Chip
      </Chip>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});