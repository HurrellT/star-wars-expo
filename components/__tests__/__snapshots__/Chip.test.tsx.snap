// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip Component renders correctly with all props combined 1`] = `
<View
  className="self-start px-4 py-2 text-base rounded-full bg-blue-500 border-blue-500  combined-chip"
>
  <Text
    className="font-medium text-white"
  >
    Combined Props Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with bordered variant 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-transparent border-slate-500 border "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Bordered Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with custom className 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-slate-500/20 border-slate-500/30  custom-chip"
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Custom Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with danger color 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-red-500/20 border-red-500/30  "
>
  <Text
    className="font-medium text-red-700 dark:text-red-200"
  >
    Danger Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with default props 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-slate-500/20 border-slate-500/30  "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Default Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with flat variant 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-slate-500/20 border-slate-500/30  "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Flat Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with ghost variant 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-transparent hover:bg-slate-500/20  "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Ghost Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with large size 1`] = `
<View
  className="self-start px-4 py-2 text-base rounded-full bg-slate-500/20 border-slate-500/30  "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Large Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with primary color 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-blue-500/20 border-blue-500/30  "
>
  <Text
    className="font-medium text-blue-700 dark:text-blue-200"
  >
    Primary Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with secondary color 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-purple-500/20 border-purple-500/30  "
>
  <Text
    className="font-medium text-purple-700 dark:text-purple-200"
  >
    Secondary Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with small size 1`] = `
<View
  className="self-start px-2 py-1 text-xs rounded-full bg-slate-500/20 border-slate-500/30  "
>
  <Text
    className="font-medium text-slate-700 dark:text-slate-200"
  >
    Small Chip
  </Text>
</View>
`;

exports[`Chip Component renders correctly with solid variant 1`] = `
<View
  className="self-start px-3 py-1.5 text-sm rounded-full bg-slate-500 border-slate-500  "
>
  <Text
    className="font-medium text-white"
  >
    Solid Chip
  </Text>
</View>
`;
